packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -162,14 +162,59 @@ export class TransformerResolver implements TransformerResolverProvider {
       slotEntry = [];
     }
 
-    slotEntry.push({
-      requestMappingTemplate,
-      responseMappingTemplate,
-      dataSource,
-    });
+    if (!this.updateSlotIfExists(slotName, requestMappingTemplate, responseMappingTemplate)) {
+      slotEntry.push({
```

Comment:
I think we can improve the understandability of this snippet here. Having `updateSlotIfExists` return control flow signals seems like one of those things we'll confuse or mix up later on. I'd prefer if we either 1) treat this method as an upsert taking the 4 params we're using in both places, or 2) have control flow helper that determines if slotExists, then a separate push/update flow that follows.

Reply:
Good catch. Refactored into 3 different functions - findSlot, updateSlot and slotExists.